pkg/util/cluster.go

```diff
@@ -426,3 +427,48 @@ func isValidCipherName(str string) bool {
 		slices.Contains(openshiftconfigv1.TLSProfiles[openshiftconfigv1.TLSProfileIntermediateType].Ciphers, str) ||
 		slices.Contains(openshiftconfigv1.TLSProfiles[openshiftconfigv1.TLSProfileModernType].Ciphers, str)
 }
+
+func getNodesCount(cl client.Client, nodesType NodesType) (int, error) {
+	if nodesType == ControlPlaneNodes {
```

Comment:
can we read the whole node list and count the items according to their annotations, instead of several reads?

Reply:
according to the node-role labels, yes. i'll update.

Reply:
done.